cli/src/commands/task/take_more_labels.rs

```diff
@@ -0,0 +1,258 @@
+                })?;
+
+            for source_annotated_comment in source_annotated_comments.into_iter() {
+                let AnnotatedComment {
+                    comment: Comment { uid, .. },
+                    labelling: source_labelling,
+                    ..
+                } = source_annotated_comment;
+                let source_assigned = source_labelling.unwrap_or_default().assigned;
+                if source_assigned.is_empty() {
+                    continue;
+                }
+                let Labelling {
+                    assigned: target_assigned,
+                    dismissed,
+                    ..
+                } = target_labellings_to_update
+                    .remove(&uid)
+                    .expect("Tried to make a new labelling for a comment we don't need to update.");
+
+                // Merge assigned labels from the two datasets
+                let assigned = merge_assigned_labels(
+                    target_assigned,
+                    source_assigned,
+                    &transferred_label_parent,
+                );
+                let new_labelling = NewLabelling {
+                    assigned,
+                    dismissed,
+                };
+
+                client
+                    .update_labelling(
+                        &target_dataset.full_name(),
+                        &uid,
+                        Some(&new_labelling),
+                        None,
+                    )
+                    .chain_err(|| ErrorKind::Client("Updating labelling failed.".into()))?;
+                statistics.add_labellings_updated(1);
+            }
+
+            let _progress = make_progress()?;
+        }
+    }
+    Ok(())
+}
+
+/// Collect target comment ids and labellings, which contain the given assigend label.
+fn get_labellings_with_assigned_label(
+    annotated_comments: Vec<AnnotatedComment>,
+    assigned_label_name: &LabelName,
+) -> HashMap<Uid, Labelling> {
+    let mut target_labellings_to_update = HashMap::new();
+    // Consume the comments here as we break them up into parts for reuse
+    for annotated_comment in annotated_comments.into_iter() {
+        let AnnotatedComment {
+            comment, labelling, ..
+        } = annotated_comment;
+        // If our target label has the label that indicates we need to take labels
+        if let Some(labelling) = labelling {
+            if labelling
+                .assigned
+                .iter()
+                .any(|label| &label.name == assigned_label_name)
+            {
+                let Comment { uid, .. } = comment;
+                target_labellings_to_update.insert(uid, labelling);
+            }
+        }
+    }
+    target_labellings_to_update
+}
+
+/// Merge the old and new assigned labels together.
+///
+/// Transferred labels will be nested under `transferred_label_parent` unless
+/// they already begin with that parent label.
+fn merge_assigned_labels(
```

Comment:
I'd like to chat about this, I have a few questions about this function but it would be easier to talk.